packages/pages/src/App.js

```diff
@@ -59,57 +66,66 @@ export default provideState({
     id: "",
   }),
   effects: {
-    async initialize() {
-      await this.effects.refreshEntries();
+    initialize() {
+      const { effects } = this;
+      websocket.onopen = async function() {
+        console.log("Connected to the server");
+        await effects.refreshEntries();
+      };
+      websocket.onerror = function(event) {
+        console.log(event);
+      };
     },
-    async refreshEntries() {
-      const response = await fetch("/api/", {
-        method: "post",
-        body: format.request(0, "listEntries", {}),
-      });
-      this.state.entries = parse(await response.text()).result;
+    refreshEntries() {
+      const { state } = this;
+      websocket.send(format.request(0, "listEntries", {}));
+      websocket.onmessage = async function(event) {
+        state.entries = await parse(event.data).result;
+      };
     },
-    async deleteEntry(
+    deleteEntry(
       _,
       {
         target: { value },
       }
     ) {
-      const response = await fetch("/api/", {
-        method: "post",
-        body: format.request(0, "deleteEntry", {
+      const { effects } = this;
+      websocket.send(
+        format.request(0, "deleteEntry", {
           id: value,
-        }),
-      });
-      try {
-        await parse.result(await response.text());
-        await this.effects.refreshEntries();
-      } catch (error) {
-        console.error(error);
-      }
+        })
+      );
+      websocket.onmessage = async function(event) {
+        try {
+          await parse.result(await event.data);
+          await effects.refreshEntries();
```

Comment:
You should no longer need to refresh entries manual after making a change, the server should notify you.